scripts/locations_data_load/locations_lambda.py

```diff
@@ -0,0 +1,283 @@
+
+    for item in processed_data:
+        identifier_value = item.get("lookup_field", {})
+
+        # Check if the identifier already exists in DynamoDB
+        if not data_exists(table, identifier_value):
+            # If the data doesn't exist, insert it into DynamoDB
+            table.put_item(Item=item)
+
+    # Call the function to update records in DynamoDB based on lookup_field
+    update_records()
+
+
+def data_exists(table, identifier_value):
+    response = table.scan(FilterExpression=Attr("lookup_field").eq(identifier_value))
+    items = response.get("Items")
+    return bool(items)
+
+
+def update_records():
+    dynamodb = boto3.resource("dynamodb")
+    org_table = dynamodb.Table("organisations")
+    locations_table = dynamodb.Table("locations")
+    org_response = org_table.scan()
+    locations_response = locations_table.scan()
+    org_items = org_response.get("Items")
+    locations_items = locations_response.get("Items")
+
+    for locations_item in locations_items:
+        locations_id = locations_item.get("id")
+        if locations_item.get("managingOrganization") == "":
+            locations_lookup_field_value = locations_item.get("lookup_field")
+
+            if locations_lookup_field_value:
+                for org_item in org_items:
+                    org_identifier_value = org_item.get("identifier", {}).get(
+                        "value", ""
+                    )
+                    if org_identifier_value == locations_lookup_field_value:
+                        org_id = org_item.get("id")
+                        locations_table.update_item(
+                            Key={"id": locations_id},
+                            UpdateExpression="SET managingOrganization = :val",
+                            ExpressionAttributeValues={":val": org_id},
+                        )
+
+
+odscode_file_path = "./ODS_Codes.xlsx"
+
+
+def read_excel_values(odscode_file_path):
+    # Read values from the Excel file
+    excel_data = pd.read_excel(odscode_file_path)
+    param1_values = excel_data["ODS_Codes"].tolist()
+
+    # # Hardcoded values for param2
+    param2_value = [
+        "OrganizationAffiliation:primary-organization",
+        "OrganizationAffiliation:participating-organization",
+    ]
+
+    # list of dictionaries with the desired format
+    params = [
+        {"primary-organization": val, "_include": param2_value} for val in param1_values
+    ]
+
+    return params
+
+
+token_api_endpoint = "https://beta.ods.dc4h.link//authorisation/auth/realms/terminology/protocol/openid-connect/token"
+
+
+# DynamoDB table name
+dynamodb_table_name = "locations"
+
+
+# def write_to_json(output_file_path, processed_data):
+#     import json
+#     with open(output_file_path, "a") as output_file:
+#         json.dump(processed_data, output_file, indent=2)
+#         output_file.write("\n")
+
+
+# # Iterate over Excel values and make API requests
+def fetch_organizations():
```

Comment:
Could be wrong but I don't think this is covered by unit tests which means we are not testing teh logic ? Could we ?